File: Sample/app/src/main/java/net/nend/sample/NativeLayoutActivity.java
package net.nend.sample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.RelativeLayout;

import net.nend.android.NendAdNative;
import net.nend.android.NendAdNativeClient;
import net.nend.android.NendAdNativeViewBinder;

public class NativeLayoutActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int spotId;
        String apiKey;
        int layout;
        int type = 0;

        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            type = bundle.getInt("type");
        }

        switch (type) {
            default:
                layout = R.layout.native_small_square;
                spotId = 485516;
                apiKey = "16cb170982088d81712e63087061378c71e8aa5c";
                break;
            case 1:
                layout = R.layout.native_large_wide;
                spotId = 485520;
                apiKey = "a88c0bcaa2646c4ef8b2b656fd38d6785762f2ff";
                break;
            case 2:
                layout = R.layout.native_text_only;
                spotId = 485522;
                apiKey = "2b2381a116290c90b936e409482127efb7123dbc";
                break;
        }

        setContentView(layout);

        final String TAG = getClass().getSimpleName();

        final NendAdNativeViewBinder binder = new NendAdNativeViewBinder.Builder()
                .adImageId(R.id.ad_image)
                .logoImageId(R.id.logo_image)
                .titleId(R.id.ad_title)
                .contentId(R.id.ad_content)
                .prId(R.id.ad_pr, NendAdNative.AdvertisingExplicitly.PR)
                .promotionUrlId(R.id.ad_promotion_url)
                .promotionNameId(R.id.ad_promotion_name)
                .actionId(R.id.ad_action)
                .build();

        final RelativeLayout adLayout = (RelativeLayout) findViewById(R.id.ad);
        NendAdNativeClient client = new NendAdNativeClient(this, spotId, apiKey);

        client.loadAd(new NendAdNativeClient.Callback() {
            @Override
            public void onSuccess(final NendAdNative nendAdNative) {
                Log.i(TAG, "広告取得成功");
                nendAdNative.intoView(adLayout, binder);
                nendAdNative.setOnClickListener(new NendAdNative.OnClickListener() {
                    @Override
                    public void onClick(NendAdNative nendAdNative) {
                        Log.i(TAG, "クリック");
                    }
                });
            }

            @Override
            public void onFailure(NendAdNativeClient.NendError nendError) {
                Log.i(TAG, "広告取得失敗 " + nendError.getMessage());
            }
        });
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/nativead/NativeLayoutActivity.kt
package net.nend.sample.kotlin.nativead

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import net.nend.android.NendAdNative
import net.nend.android.NendAdNativeClient
import net.nend.android.NendAdNativeListener
import net.nend.android.NendAdNativeViewBinder
import net.nend.sample.kotlin.R
import net.nend.sample.kotlin.nativead.NativeSampleActivity.Companion.NATIVE_API_KEY_LARGE_WIDE
import net.nend.sample.kotlin.nativead.NativeSampleActivity.Companion.NATIVE_API_KEY_SMALL_SQUARE
import net.nend.sample.kotlin.nativead.NativeSampleActivity.Companion.NATIVE_API_KEY_TEXT_ONLY
import net.nend.sample.kotlin.nativead.NativeSampleActivity.Companion.NATIVE_SPOT_ID_LARGE_WIDE
import net.nend.sample.kotlin.nativead.NativeSampleActivity.Companion.NATIVE_SPOT_ID_SMALL_SQUARE
import net.nend.sample.kotlin.nativead.NativeSampleActivity.Companion.NATIVE_SPOT_ID_TEXT_ONLY

class NativeLayoutActivity : AppCompatActivity() {

    private enum class NativeType(val id: Int) {
        LARGE(1),
        TEXT_ONLY(2)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val spotId: Int
        val apiKey: String
        val layout: Int
        val type = intent.extras?.getInt("type") ?: return

        when (type) {
            NativeType.LARGE.id -> {
                layout = R.layout.native_large_wide
                spotId = NATIVE_SPOT_ID_LARGE_WIDE
                apiKey = NATIVE_API_KEY_LARGE_WIDE
            }
            NativeType.TEXT_ONLY.id -> {
                layout = R.layout.native_text_only
                spotId = NATIVE_SPOT_ID_TEXT_ONLY
                apiKey = NATIVE_API_KEY_TEXT_ONLY
            }
            else -> {
                layout = R.layout.native_small_square
                spotId = NATIVE_SPOT_ID_SMALL_SQUARE
                apiKey = NATIVE_API_KEY_SMALL_SQUARE
            }
        }

        setContentView(layout)

        val binder = NendAdNativeViewBinder.Builder()
                .adImageId(R.id.ad_image)
                .logoImageId(R.id.logo_image)
                .titleId(R.id.ad_title)
                .contentId(R.id.ad_content)
                .prId(R.id.ad_pr, NendAdNative.AdvertisingExplicitly.PR)
                .promotionUrlId(R.id.ad_promotion_url)
                .promotionNameId(R.id.ad_promotion_name)
                .actionId(R.id.ad_action)
                .build()

        NendAdNativeClient(this, spotId, apiKey).run {
            loadAd(object : NendAdNativeClient.Callback {
                override fun onSuccess(nendAdNative: NendAdNative) {
                    Log.i(NativeSampleActivity.NATIVE_LOG_TAG, "広告取得成功")
                    nendAdNative.intoView(findViewById(R.id.ad), binder)
                    nendAdNative.setNendAdNativeListener(object : NendAdNativeListener {
                        override fun onImpression(nendAdNative: NendAdNative) {
                            Log.i(NativeSampleActivity.NATIVE_LOG_TAG, "onImpression")
                        }

                        override fun onClickAd(nendAdNative: NendAdNative) {
                            Log.i(NativeSampleActivity.NATIVE_LOG_TAG, "onClickAd")
                        }

                        override fun onClickInformation(nendAdNative: NendAdNative) {
                            Log.i(NativeSampleActivity.NATIVE_LOG_TAG, "onClickInformation")
                        }
                    })
                }

                override fun onFailure(nendError: NendAdNativeClient.NendError) {
                    Log.i(NativeSampleActivity.NATIVE_LOG_TAG, "広告取得失敗: ${nendError.message}")
                }
            })
        }
    }
}